scene/gui/button.cpp

```diff
@@ -371,10 +410,15 @@ void Button::_shape(Ref<TextParagraph> p_paragraph, String p_text) {
 		p_text = xl_text;
 	}
 
-	Ref<Font> font = get_theme_font(SNAME("font"));
-	int font_size = get_theme_font_size(SNAME("font_size"));
-
 	p_paragraph->clear();
+
+	Ref<Font> font = theme_cache.font;
+	int font_size = theme_cache.font_size;
+	if (font.is_null() || font_size == 0) {
```

Comment:
Those are not checks, though. Those are errors. And in this particular case I want to avoid throwing errors.

The problem this snippet solves is that buttons set text in their constructor. But at that point we don't have the cache yet. The cache is first set up on `NOTIFICATION_POSTINITIALIZE`. I didn't find a better way to do this for all controls as early as possible without manually running the method in each constructor, which would be quite awful. So for now, when `Button()` is constructed it calls `set_text` and hits these errors. Hence the extra check here. Didn't find any such issues with other controls.

It should shape correctly upon entering the tree anyway, so this check is only to silence the error from `text_paragraph.cpp`.

Reply:
OK, if it's to suppress the errors it seems fine.